app/src/main/java/com/example/tasky/agenda/data/local/model/relations/EventPhotoCrossReference.kt

```diff
@@ -0,0 +1,9 @@
+package com.example.tasky.agenda.data.local.model.relations
+
+import androidx.room.Entity
+
+@Entity(primaryKeys = ["id", "key"])
+data class EventPhotoCrossReference(
+    val id: String,
+    val key: String
+)
```

Comment:
Do you really need this? The event-photo relation should be a 1-to-many relation and not a many-to-many relation. One event can have multiple photos, but one photo can only belong to one event (Each photo will get a different key/url).
Therefore, no cross ref is needed here, since you just need to save the eventId for each photo. That will be enough to know which event contains which photos and which photo belongs to which event.

Reply:
I think he did many-to-many, in case a user uploads the same photo to multiple events?


So it makes sense for that usecase
Kinda overkill for this type of app, but still makes sense there

Reply:
No, since in the context of Tasky and the backend 2 photos aren't considered the same when they have the same pixels, but when they have the same URL and key. And this is not considered by the backend - if you upload the same photo twice across events, you will get back 2 photos with different keys, even though they're the same photo.